src/analysis_file.py

```diff
@@ -82,7 +81,8 @@ def generate(user, data, csv_by_message_output_path, csv_by_individual_output_pa
                             analysis_dict[f"{cc.analysis_file_key}{code_string_value}"] = Codes.MATRIX_1
 
                         for key in show_matrix_keys:
-                            if key not in analysis_dict:
+                            if key not in analysis_dict and (key.endswith(Codes.STOP) or key.endswith(Codes.TRUE_MISSING)
```

Comment:
Could you add a comment explaining why there are exceptions given for STOP, NA, and NC?

I don't think it's obvious, as you have experienced first hand :P

Reply:
If there was a way to structure this so we could refer to the control_code field of the code objects rather than doing a string endswith, that would be a bit cleaner/safer. For example, you could generate a list of exceptions when iterating through the codes on L76, then check the key for membership of the exceptions list here.

However, as this is a local fix for IMAQAL-only, I'm not too fussed.

Reply:
Todo thanks.